### DISCUSSION TITLE
help me implement the "login with google" system

### DISCUSSION THREAD

**Discussion by joaopedronunesm** [community/community#140809] [Programming Help]
### Body

help me translate the codes that are in kotlin in the google documentation to java, I have already looked for programs and I was not able to successfully translate the site is https://developer.android.com/identity/sign-in/credential-manager-siwg

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by VeerammalV** [comment]
Create a OAuth credentials in Google cloud console using your App name, package name, SHA-1 fingerprint and enable the API which you wanted for your project. And download the json file and add it in your project. The old method of Google sign in was deprecated but still it works. The latest one is Credential Manager.. I'm facing an issue with this credential manager. I've setup my OAuth credentials and added the json file to my project but Google is not signing in. I found that in some documentation it was suggested that we have to use WebClient Id and not Android Client Id.. Still finding ways to get resolve this issue.. if you use Android studio, the kotlin code will be cancelled be changed to java.. Make sure you file format is .java file. Android studio itself converts the code from kotlin to java.


Here the old version of Google sign in : [https://github.com/VeerammalV/Event-Schedular-Google-Calendar-UI](url)

**Reply by joaopedronunesm** [reply]
but i am not using android studio
